### PR TITLE
Normalize indentation and apply basic PSR2 CS ruleset

### PR DESCRIPTION
**Pull request by mvorisek**
improve https://github.com/code-lts/WarnockPDF/pull/4
partly https://github.com/code-lts/WarnockPDF/pull/3

there should be no code change, thus fully compatible with PHP5

### CODE REVIEW

File: .php_cs.dist [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
```
Comment (williamdes) [inline .php_cs.dist:1]: @mvorisek did you see there is a phpcs.xml configuration file?

Comment (williamdes) [reply .php_cs.dist:1]: Only if the fixer can read the phpcs.xml format ;)

Comment (mvorisek) [reply .php_cs.dist:1]: With a few manual fixes, I managed to make both CS fixers happy. CI added as well.

Comment (mvorisek) [reply .php_cs.dist:1]: With a few manual fixes, I managed to make both CS fixers happy. CI added as well.

File: .github/workflows/lint-and-analyse.yml [hunk 2]
```diff
@@ -47,3 +66,14 @@ jobs:
...
+    doctum:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+            - name: lint php documentation
+              uses: sudo-bot/action-doctum@dev
+              with:
+                  config-file: scripts/doctum.php
+                  method: "parse"
+                  cli-args: "--output-format=github --no-ansi --no-progress -v --ignore-parse-errors"
```
Comment (williamdes) [inline .github/workflows/lint-and-analyse.yml:79]: why did you move it here ?

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,7 @@
 .vscode
 .idea
 phpstan.neon
-vendor
\ No newline at end of file
+.php_cs
+.php_cs.cache
```
Comment (williamdes) [inline .gitignore:5]: I will need to add this to the phpcs.xml file

**Review by williamdes — APPROVED** [review]
`.php_cs.dist` needs to be added to the `.gitattributes` file
Nice work

File: .github/workflows/lint-and-analyse.yml (hunk 2 above)
Comment (mvorisek) [reply .github/workflows/lint-and-analyse.yml:79]: image

to display everything on one page

File: .gitignore (hunk 3 above)
Comment (mvorisek) [reply .gitignore:5]: These are files, I belive phpcs fixes only `.php` files, thus I belive it is not needed.

Comment (williamdes) [reply .gitignore:5]: I was thinking about `<arg name="cache" value=".phpcs-cache"/>` :)

File: .github/workflows/lint-and-analyse.yml (hunk 2 above)
Comment (williamdes) [reply .github/workflows/lint-and-analyse.yml:79]: Okay, but please avoid such un-related changes :)

**Comment by williamdes** [comment]
I removed the cs tool changes from the other tool because I tested it and was not convinced. This project will stick to phpcs, It works great as always.

Thanks for your work, I merged it 🚀

**Comment by mvorisek** [comment]
I am disappointed by this update. The changes I proposed here were for indentation and array CS as well.

> I tested it and was not convinced.
What is wrong by using CS Fixer tool as well? Can I address that somehow?

The issue I have now is fork that does have already a lot of changes, but not enought to keep with current PSR12 style. I already maintain a fork of TCPDF (internal only) and with these changes, I can not smoothly merge it with the original project nor this project.

**Comment by williamdes** [comment]
> I am disappointed by this update. The changes I proposed here were for indentation and array CS as well.
> 
> > I tested it and was not convinced.
> > What is wrong by using CS Fixer tool as well? Can I address that somehow?
> 
> The issue I have now is fork that does have already a lot of changes, but not enought to keep with current PSR12 style. I already maintain a fork of TCPDF (internal only) and with these changes, I can not smoothly merge it with the original project nor this project.
> 
> Please revert to [8e9ca23](https://github.com/code-lts/WarnockPDF/commit/8e9ca23f04c54450e6a59d6c7859780021813a06) and force push. phpcs is welcomed to stay, feel free to add the new rules there as well.

I understand that you can be des appointed, I do not like the feeling of have the cs fixer because I do not use it anywhere else.
That said, I would have been happy to know that you want the PSR12 changes to make a fork of my fork.
I am pretty sure you need to open source them because of the gpl license.

Anyway, I will apply more of the PSR12 ruleset.
You can still help me, but you need to use the actual tool.

So please tell me your objectives rather than trying to secretly make some changes pass.

We can not build great software if everybody has secret objectives doing forks of forks.
Maybe you have very cool changes internally.
I would be very happy to welcome them.

Please let me know your thoughts

**Comment by mvorisek** [comment]
Yesterday, I put a lot of efford to fix the most LOC with current standards (`fonts/` as well, that is code too!, it took me 2 hours to help the fixer to finish in reasonable time). Basically, I kept out short array syntax and some minor cosmetical changes that can be applied by higher level of PHP rules.

I use (and contribute/report to) CS Fixer because I find it:
- more and more used by the most popular projects
- the rules are much more simpler - they do a very good job with their rulesets, basically the whole project CS can be defined with one, two ruleset and optionally several extra rules/excludes

So for easier maintanance, I need a code that can comply with CS Fixer. I achived that yesterday with a few iterations with you.

Today, I woke up and all changes are gone with 1st commit that I did not even authored not agree with (at least https://github.com/code-lts/WarnockPDF/pull/5/commits/2f6cd85b60b7143709837271699a0d382c0a3d49#diff-05ae9cddcaec1e845771a7db224961439f83ef5939ec67d3a48744cb34d7e58bR10 is completely wrong to me as this file is used by CS Fixer not phpcs).

So I propose to revert to https://github.com/code-lts/WarnockPDF/commit/8e9ca23f04c54450e6a59d6c7859780021813a06, force push and then increase the rules in phpcs and possibly in CS Fixer as well.

**Comment by williamdes** [comment]
> So I propose to revert to 8e9ca23, force push and then increase the rules in phpcs and possibly in CS Fixer as well.

I can not revert it, it is not part of the history.

> Yesterday, I put a lot of efford to fix the most LOC with current standards (fonts/ as well, that is code too!, it took me 2 hours to help the fixer to finish in reasonable time). Basically, I kept out short array syntax and some minor cosmetical changes that can be applied by higher level of PHP rules.

I apologise for the time you had to spend, I also had to spend time. And that was 5 hours of my normal night time.

> Today, I woke up and all changes are gone with 1st commit that I did not even authored not agree with

Okay, I pushed c79c77c to change the name

> So for easier maintenance, I need a code that can comply with CS Fixer. I achieved that yesterday with a few iterations with you.

You can pick your ruleset and let me know the rules that re not passing. I will do the changes on the phpcs tool :)


What I think you can do is open a new PR with code changes only, rebased on the latest cs fixes.
If I understand your ruleset can just be copied on top of latest main branch and ran, that should produce the same fixed files.
I believe I merged most of your changes

Let me know, and be sure I value your contribution and time spent.

**Comment by mvorisek** [comment]
> I can not revert it, it is not part of the history.

I meant to hard reset to.

I will try to find some more time to submit another PR.

To prevent this happening again, please:
- NEVER FORCE PUSH so radical changes to someone's PR
- and let me know if you are ok with fixing `fonts/` php files as well
- adding CS Fixer to CI (at least with some basic rules) if it can coexist with phpcs

**Comment by williamdes** [comment]
> I meant to hard reset to.

The history has more commits now, I can not reset. Hours of work fixing cs rules and phpdoc blocks.

> adding CS Fixer to CI (at least with some basic rules) if it can coexist with phpcs

I am not in favor of this, I want to stick to the actual phpcs tool :/

> and let me know if you are ok with fixing fonts/ php files as well

They are generated files I think, there is some code in `tools` that seems to generate them.
But anyway you can format them.


> NEVER FORCE PUSH so radical changes to someone's PR

Sorry for that, but I would just ask that next time you try to make different commits as small as possible to be able to audit the changes before merging. For example one for the configs and one for the changes.

> I will try to find some more time to submit another PR.

Thank you !

**Comment by mvorisek** [comment]
I managed now to knock out the original history with `git pull origin 4653ae281a02c51225f3df168ccad61f59c3514c`.

> I am not in favor of this, I want to stick to the actual phpcs tool :/

Diffing the changes they are still a lot of diffs...

a) why are you against two CS fixers which can coexist?
b) phpcs seems too slow to process the fonts/, see https://github.com/code-lts/WarnockPDF/pull/8/checks?check_run_id=2201526345
c) please update the phpcs rules to:
  i. to not add space between `array` and `(`, ie. to norm array opening to `array(`
  ii. put opening braces of method bodies on a NL (required by PSR-12)
